src/js/jsx/sections/libraries/LibraryList.jsx

```diff
@@ -71,10 +72,23 @@ define(function (require, exports, module) {
                 command: null
             };
         },
+        
+        shouldComponentUpdate: function (nextProps, nextState) {
+            return this.props.libraries !== nextProps.libraries ||
+                   this.props.selected !== nextProps.selected ||
+                   this.props.disabled !== nextProps.disabled ||
+                   this.props.onLibraryChange !== nextProps.onLibraryChange ||
+                   !_.isEqual(this.state, nextState);
+        },
 
         componentDidUpdate: function () {
-            if (this.refs.input) {
-                this.refs.input.getDOMNode().focus();
+            var renameInput = this.refs.input;
+            
+            if (renameInput) {
+                window.setTimeout(function () {
+                    renameInput.acquireFocus();
+                    renameInput.getDOMNode().focus();
+                }, 100);
```

Comment:
For some reason, using `element.getDOMNode().focus()` alone does not focus the rename input property. It only select the text but not in a focused state (without showing the blue bottom border). I am not confident that using `setTimeout` and `acquireFocus` is a good solution. So anyone who worked on the Focusable component before, please let me know your thoughts.